### PR TITLE
feat(git): generate from specific tag

### LINKED ISSUE #27: Support commit ranges
Creating changelogs requires [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages) and the execution of `changelog -M`, `changelog -m` or `changelog -p`. Is there also possibility to create a changelog from a commit range? 

I am thinking of something like: `changelog 420c945...2a83752`.

GitHub supports showing a comparison of two SHA1 hashes, like:
- https://github.com/welovecoding/website/compare/420c945...2a83752

### PR DESCRIPTION
**Pull request by ffflorian**
This pull request lets the user generate a changelog back from a specific commit, across branches and within commit ranges. Fixes #27.

Example: `generate -p 420c945...2a83752` or `generate -p master..new-branch`.

This PR is still missing tests but unfortunately I don't really know how to write them since I am not very experienced with Sinon and wouldn't know how to test the CLI. Maybe @robinjoseph08 can help?

### CODE REVIEW

**Comment by tvardy** [comment]
Hi! I can have a look today and add test coverage. As I'd also like to use this feature in my projects ;)

**Comment by tvardy** [comment]
@ffflorian smal fix and test coverage sent as PR to your fork ;)

**Comment by ffflorian** [comment]
@tvardy Awesome, I'll have a look :)

**Comment by bennyn** [comment]
No pressure but is there already a plan when to merge it? :)

**Comment by tvardy** [comment]
There's still a small fix of mine pending on @ffflorian fork. After that, it's up to LOB team to check and decide.

**Comment by ffflorian** [comment]
Done, sorry for the delay!

File: lib/cli.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ module.exports = CLI
   .option('-p, --patch', 'create a patch changelog')
   .option('-m, --minor', 'create a minor changelog')
   .option('-M, --major', 'create a major changelog')
+  .option('-t, --tag [range]', 'generate from specific tag', '')
```
Comment (robinjoseph08) [inline lib/cli.js]: Can you change it from `[range]` to `<range>`? That will indicate that if you specify `-t`, a range is [a required argument](https://www.npmjs.com/package/commander#specify-the-argument-syntax). And then you can remove the default empty string. If someone is using the flag, they should be passing something in.

Also, can you add an example of a normal tag and then a range in the description? `generate from specific tag or range e.g. v1.2.3, v1.2.3..v1.2.4`

File: README.md [hunk 2]
```diff
@@ -58,6 +58,7 @@ $ changelog -h
     -p, --patch            create a patch changelog
     -m, --minor            create a minor changelog
     -M, --major            create a major changelog
+    -t, --tag              generate from specific tag
```
Comment (robinjoseph08) [inline README.md]: Once you've updated `cli.js`, can you re-run the help and put the updated output here?

Comment (ffflorian) [reply README.md]: The updated help text:
```
$ bin/generate --help

  Usage: generate [options]

  Generate a changelog from git commits.


  Options:

    -V, --version          output the version number
    -p, --patch            create a patch changelog
    -m, --minor            create a minor changelog
    -M, --major            create a major changelog
    -t, --tag <range>      generate from specific tag or range (e.g. v1.2.3 or v1.2.3..v1.2.4)
    -x, --exclude <types>  exclude selected commit types (comma separated)
    -f, --file [file]      file to write to, defaults to ./CHANGELOG.md, use - for stdout
    -u, --repo-url [url]   specify the repo URL for commit links, defaults to checking the package.json
    -h, --help             output usage information
```

Does it look okay?

Comment (robinjoseph08) [reply README.md]: Yup!

**Comment by bennyn** [comment]
Great! Can't wait to see it in an upcoming release. 🤘

**Comment by robinjoseph08** [comment]
Just published v1.5.0! Thanks again for all of your help!

**Comment by ffflorian** [comment]
Great 🎉

**Pull request merged**